src/server/protocol.ts

```diff
@@ -563,7 +563,8 @@ namespace ts.server.protocol {
         arguments: GetApplicableRefactorsRequestArgs;
     }
     export type GetApplicableRefactorsRequestArgs = FileLocationOrRangeRequestArgs & {
-        triggerReason?: RefactorTriggerReason
+        triggerReason?: RefactorTriggerReason;
+        refactorKind?: string;

```

Comment:
For the protocol, will the `kind` also be returned on the code action objects?

Reply:
Good catch. 

Also `refactorKind` does seem a bit redundant. Would just `kind` meaningful enough?